snapcraft/parts/parts.py

```diff
@@ -143,6 +143,23 @@ def prime_dir(self) -> pathlib.Path:
         """Return the parts prime directory path."""
         return self._lcm.project_info.prime_dir
 
+    @property
+    def prime_dirs(self) -> dict[str | None, pathlib.Path]:
```

Comment:
this should probably replace def get_prime_dir_for_component
this is also needs to be renames as it is not all prime_dirs, it is missing the default, so component_prime_dirs would be more accurate